src/extra/gpkg.jl

```diff
@@ -0,0 +1,1088 @@
+      else
+        return Point(crs_type(x, y, coords_list[3]))
+      end
+    end
+  catch e
+    # If CRS construction fails, fall back to raw coordinates
+    @warn "Failed to create point with CRS $crs_type: $e. Using raw coordinates."
+    return Point(coords_list...)
+  end
+end
+
+function parse_wkb_point(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool, crs_type)
+  coords_list = read_coordinate_values(io, byte_order, has_z, has_m)
+  return create_point_from_coords(coords_list, crs_type)
+end
+
+function parse_wkb_linestring(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool, crs_type; as_rope::Bool=false)
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  
+  num_points = read_value(read(io, UInt32))
+  
+  if num_points == 0
+    return nothing
+  end
+  
+  # Parse all points using helper functions
+  points = Point[]
+  
+  for i in 1:num_points
+    coords_list = read_coordinate_values(io, byte_order, has_z, has_m)
+    point = create_point_from_coords(coords_list, crs_type)
+    if !isnothing(point)
+      push!(points, point)
+    end
+  end
+  
+  if isempty(points)
+    return nothing
+  end
+  
+  # If forced to be a Rope (for MultiLineString consistency) or open linestring
+  if as_rope || length(points) < 2 || points[1] != points[end]
+    return Rope(points...)
+  else
+    # It's a closed linestring, return a Ring (exclude the duplicate last point)
+    return Ring(points[1:end-1]...)
+  end
+end
+
+function parse_wkb_ring(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool, crs_type)
+  # Parse ring by temporarily reading as linestring and converting to Ring
+  # First read the number of points
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  num_points = read_value(read(io, UInt32))
+  
+  if num_points == 0
+    return nothing
+  end
+  
+  # Parse all points
+  points = Point[]
+  for i in 1:num_points
+    coords_list = read_coordinate_values(io, byte_order, has_z, has_m)
+    point = create_point_from_coords(coords_list, crs_type)
+    if !isnothing(point)
+      push!(points, point)
+    end
+  end
+  
+  if isempty(points)
+    return nothing
+  end
+  
+  # For rings, exclude duplicate last point if present (WKB rings are closed with duplicate point)
+  if length(points) >= 2 && points[1] == points[end]
+    return Ring(points[1:end-1]...)
+  else
+    # If not closed with duplicate, still return Ring (assuming it should be closed)
+    return Ring(points...)
+  end
+end
+
+function parse_wkb_polygon(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool, crs_type)
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  
+  num_rings = read_value(read(io, UInt32))
+  
+  if num_rings == 0
+    return nothing
+  end
+  
+  # Parse exterior ring using ring parsing function
+  exterior_ring = parse_wkb_ring(io, byte_order, has_z, has_m, crs_type)
+  if isnothing(exterior_ring)
+    return nothing
+  end
+  
+  # Parse interior rings (holes)
+  hole_rings = Ring[]
```

Comment:
Same issue with abstract type `Ring[]` initialization. Use the map-do syntax to avoid manual initialization of types.